src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -156,27 +154,32 @@ public virtual void Abort()
                             {
                                 if (NegotiationProtocol.TryParseMessage(buffer, out var negotiationMessage, out consumed, out examined))
                                 {
-                                    var protocol = protocolResolver.GetProtocol(negotiationMessage.Protocol, supportedProtocols, this);
+                                    Protocol = protocolResolver.GetProtocol(negotiationMessage.Protocol, supportedProtocols, this);
 
                                     var transportCapabilities = Features.Get<IConnectionTransportFeature>()?.TransportCapabilities
                                         ?? throw new InvalidOperationException("Unable to read transport capabilities.");
 
-                                    var dataEncoder = (protocol.Type == ProtocolType.Binary && (transportCapabilities & TransferMode.Binary) == 0)
-                                        ? (IDataEncoder)Base64Encoder
-                                        : PassThroughEncoder;
+                                    var requiredTransferMode = (Protocol.Type == ProtocolType.Binary) ?
+                                        TransferFormat.Binary :
+                                        TransferFormat.Text;
+
+                                    if((transportCapabilities & requiredTransferMode) == 0)
```

Comment:
nit: space after if

Reply:
Get used to this. His keyboard is broken 😆

Reply:
For some reason VS doesn't autoformat very well with VsVim in place (and at this point I <3 VsVim too much to give it up). I keep meaning to look in to fixing it or building something to automatically reformat before I commit, but the former means learning F# (\*grumble\*) and I've been too lazy to do the latter. @davidfowl has seemed content to just point them all out for me conveniently, so.... 😝

I might finally take a look at that auto-formatter this weekend...